# Introduction and Conceptual Explanation

Louise’s strategy is an elementary form of technical analysis known as market timing. This is subjective, as traders may interpret price changes differently, and have different rules depending on their risk aversion. It is difficult to time entrances and exits. Research by William Sharpe concluded that an investor using a market timing strategy must be correct 74% of the time to beat the benchmark portfolio of similar risk on an annual basis, which is difficult to accomplish. If Louise inadvertently moves her portfolio out of VFINX during the best months, then she will be losing more than if she held it for the entire duration passively, incurring an opportunity cost. There is also research by market analysts that conclude that a passively managed portfolio will outperform an actively managed portfolio that seeks to time the market. Another consideration is transaction costs and commissions, which are higher due to the frequent movement of portfolio funds. Because the security was held less than a year, Louise also would be taxed at short term capital gains, which are higher than the long term capital gains rate.

In addition, according to the efficient markets hypothesis, share prices reflect all available information, and stocks always trade at their fair value on exchanges. Therefore, it would be impossible for investors to outperform the market. The weak form of the theory assumes that current stock prices reflect all available information and that historical performance is irrelevant to future performance. As a result, the market timing technique used by Louise cannot be used to generate returns because past prices are already reflected in today’s stock price. 

One of the first things that we remember when we invest is the principle of diversification. Louise’s strategy or portfolio does not follow the principle of diversification. Louise shifts 100% of her portfolio based on how the market is performing. She tends to put ‘all eggs in one basket’ which is not ideal according to portfolio theory. Ideally Louise should be diving her portfolio by weight into equity and risk-free debt based on her risk appetite. If she has a high-risk high-return strategy she should place the majority of her portfolio in equity. Her portfolio should always ideally be on the Capital Market Line(CAL) of the efficient frontier.


In [1]:
from IPython.display import Image
Image(url= "Efficient Frontier Image.png")

As we can see from the above diagram, individual portfolios do not lie one on the Capital Market line and individual portfolios are not optimised. Louise’s portfolio would be one of these points that is not on the line. If she optimised her portfolio based on her risk appetite her portfolio would probably be one of the 3 points on the Efficient frontier. The Capital Market Line is always tangent to the efficient frontier. A high risk high return portfolio would lie higher up the efficiency as the standard deviation and expected return increases. 


In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt 
import plotly.offline as plyo 
import cufflinks as cf 
plyo.init_notebook_mode(connected=True)  

In [3]:
vfinx_n = pd.read_csv('VFINX.csv', index_col='Date', parse_dates=True)
vfinx_n = vfinx_n.resample('1M').last()[['Adj Close']].pct_change().multiply(100).dropna()

vfinx_n.columns = ['R']
vfinx_n

R
Date                 
1985-02-28   1.380961
1985-03-31  -0.845488
1985-04-30   0.571295
1985-05-31   6.033273
1985-06-30   0.627205
...               ...
2020-10-31  -2.276990
2020-11-30  10.935640
2020-12-31   3.441073
2021-01-31  -0.642105
2021-02-28   4.261644

[433 rows x 1 columns]

In [4]:
ff_n = pd.read_csv(
    'F-F_Research_Data_Factors.CSV',
    index_col=0,
    skiprows=3,
    nrows=12*(2020 - 1927 + 1) + 6 + 1
)
ff_n.index = pd.to_datetime(ff_n.index, format='%Y%m') + pd.offsets.MonthEnd(0)
ff_n['Date'] = pd.to_datetime(ff_n.index, format='%Y%m') + pd.offsets.MonthEnd(0)

In [5]:
df_n = vfinx_n.join(ff_n, how='inner')
df_n['Return'] = df_n['R'] / 100
df_n

R  Mkt-RF   SMB   HML    RF       Date    Return
1985-02-28   1.380961    1.22  0.76 -0.13  0.58 1985-02-28  0.013810
1985-03-31  -0.845488   -0.84 -1.15  4.10  0.62 1985-03-31 -0.008455
1985-04-30   0.571295   -0.96  0.14  3.72  0.72 1985-04-30  0.005713
1985-05-31   6.033273    5.09 -2.25 -0.92  0.66 1985-05-31  0.060333
1985-06-30   0.627205    1.27  0.50  0.50  0.55 1985-06-30  0.006272
...               ...     ...   ...   ...   ...        ...       ...
2020-09-30  -4.393229   -3.63  0.06 -2.51  0.01 2020-09-30 -0.043932
2020-10-31  -2.276990   -2.10  4.44  4.03  0.01 2020-10-31 -0.022770
2020-11-30  10.935640   12.47  5.48  2.11  0.01 2020-11-30  0.109356
2020-12-31   3.441073    4.63  4.81 -1.36  0.01 2020-12-31  0.034411
2021-01-31  -0.642105   -0.04  7.18  2.85  0.00 2021-01-31 -0.006421

[432 rows x 7 columns]

In [6]:
import numpy as np
import pandas as pd
from pandas import DataFrame
import matplotlib.pyplot as plt

pd.plotting.register_matplotlib_converters()

gspc = pd.read_csv('GSPC.csv', index_col='Date', parse_dates=True).resample('M').last().loc[:'2021-01']
gspc['Return'] = (gspc['Close'] - gspc['Open'])
gspc['Percent Return'] = ((gspc['Close'] / gspc['Open']) - 1) * 100
gspc['Sign'] = np.sign(gspc['Return'])

vfinx = pd.read_csv('VFINX.csv', index_col='Date', parse_dates=True).resample('M').last().loc[:'2021-01']
vfinx['Return'] = (vfinx['Close'] - vfinx['Open'])
vfinx['Percent Return'] = ((vfinx['Close'] - vfinx['Open']) / vfinx['Open']) * 100

ff = pd.read_csv(
    'F-F_Research_Data_Factors.CSV',
    index_col=0,
    skiprows=3,
    nrows=12*(2020 - 1927 + 1) + 6 + 1
)
ff.index = pd.to_datetime(ff.index, format='%Y%m') + pd.offsets.MonthEnd(0)
ff["Date"] = pd.to_datetime(ff.index, format='%Y%m') + pd.offsets.MonthEnd(0)


pastMonthSign = 0
pastMonthSign2 = 0
portfolio = "vfinx"
portfolioStocks = []
for monthlySign in gspc['Sign']:
    if pastMonthSign2 == 1 and pastMonthSign == 1:
        portfolio = "vfinx"
    elif pastMonthSign2 == -1 and pastMonthSign == -1:
        portfolio = "ff"
    pastMonthSign2 = pastMonthSign
    pastMonthSign = monthlySign
    portfolioStocks.append(portfolio)

finalFF = []
for i in range(len(ff["RF"])):
    if i > (len(ff["RF"]) - 433 - 1):
        finalFF.append(ff["RF"][i])
        
vfinx = vfinx.resample('1M').last()[['Adj Close']].pct_change().multiply(100).dropna()
vfinx.columns = ['R']
df = vfinx.join(ff, how='inner')
df['Return'] = df['R'] / 100
print(df)

currentBal = 0
currentBalLst = []
returnsLst = []
for i in range(len(df['Return'])):
    currentBal += 500
    if portfolioStocks[i] == "vfinx":
        currentBal = currentBal * (1 + (df['Return'][i]))
        returnsLst.append(df['R'][i])
    elif portfolioStocks[i] == "ff":
        currentBal = currentBal * (1 + finalFF[i] / 100)
        returnsLst.append(finalFF[i])
    currentBalLst.append(currentBal)

currentBalVfinx = 0
vfinxLst = []
for i in range(len(df['Return'])):
    currentBalVfinx += 500
    currentBalVfinx = currentBalVfinx * (1 + (df["Return"][i]))
    vfinxLst.append(currentBalVfinx)
    

rrfLst = []
for i in range(len(returnsLst)):
    temp = returnsLst[i] - finalFF[i]
    rrfLst.append(temp)

louiseDf = pd.DataFrame(rrfLst,columns=['R-RF'])
louiseDf['Date'] = pd.date_range(start='19850228', periods=432, freq='M')
louiseDf.set_index('Date')

                    R  Mkt-RF   SMB   HML    RF       Date    Return
1985-02-28   1.380961    1.22  0.76 -0.13  0.58 1985-02-28  0.013810
1985-03-31  -0.845488   -0.84 -1.15  4.10  0.62 1985-03-31 -0.008455
1985-04-30   0.571295   -0.96  0.14  3.72  0.72 1985-04-30  0.005713
1985-05-31   6.033273    5.09 -2.25 -0.92  0.66 1985-05-31  0.060333
1985-06-30   0.627205    1.27  0.50  0.50  0.55 1985-06-30  0.006272
...               ...     ...   ...   ...   ...        ...       ...
2020-09-30  -4.393229   -3.63  0.06 -2.51  0.01 2020-09-30 -0.043932
2020-10-31  -2.276990   -2.10  4.44  4.03  0.01 2020-10-31 -0.022770
2020-11-30  10.935640   12.47  5.48  2.11  0.01 2020-11-30  0.109356
2020-12-31   3.441073    4.63  4.81 -1.36  0.01 2020-12-31  0.034411
2021-01-31  -0.642105   -0.04  7.18  2.85  0.00 2021-01-31 -0.006421

[432 rows x 7 columns]


R-RF
Date                 
1985-02-28   0.730961
1985-03-31  -1.425488
1985-04-30  -0.048705
1985-05-31   5.313273
1985-06-30   0.000000
...               ...
2020-09-30  -4.403229
2020-10-31  -2.286990
2020-11-30  10.925640
2020-12-31   0.000000
2021-01-31   0.000000

[432 rows x 1 columns]

# Excess Return Calculations

In [7]:
df_n['R-RF'] = df_n['R'] - df_n['RF']

In [8]:
df_n['R-RF'] = df_n['R'] - df_n['RF']

# Rolling Sharpe Ratios

In [9]:
df_n['naive_rolling'] = df_n['R-RF'].rolling(12).mean() / df_n['R-RF'].rolling(12).std() * np.sqrt(12) 
df_n['naive_rolling'].dropna()
sharpe_plot = df_n[['Date','naive_rolling']]
sharpe_plot.set_index('Date')

naive_rolling
Date                     
1985-02-28            NaN
1985-03-31            NaN
1985-04-30            NaN
1985-05-31            NaN
1985-06-30            NaN
...                   ...
2020-09-30       0.652793
2020-10-31       0.454900
2020-11-30       0.701709
2020-12-31       0.739108
2021-01-31       0.702721

[432 rows x 1 columns]

# Data Driven Explanation

The Sharpe Ratio measures an investment’s returns relative to the risk. A higher sharpe ratio is considered better as it indicates optimisation and high return relative to the risk. If we analyse the chart below, we notice that the sharpe ratio is higher for the naive strategy than it is for Louise's strategy indicating that the return is better relative to the risk. This enforces our explanation of Louise’s strategy not being on the efficient frontier as it is not risk-return optimised. 


In [10]:
louiseDf['louise_rolling'] = louiseDf['R-RF'].rolling(12).mean() / louiseDf['R-RF'].rolling(12).std() * np.sqrt(12) 
louiseDf['louise_rolling'].dropna()
plotdf = louiseDf[['Date','louise_rolling']]
plotdf.set_index("Date")
finalplot = pd.merge(plotdf,sharpe_plot,on = 'Date',how = 'inner')
finalplot = finalplot.set_index("Date")
print(finalplot)
plyo.iplot(  
    finalplot[['naive_rolling','louise_rolling']].iplot(asFigure=True,
                                                 title='Sharpe Ratio Comparison',  
             xTitle='Date',  
             yTitle='Sharpe Ratio',
                                                 mode={'naive_rolling': 'lines+markers', 'louise_rolling': 'lines+markers'},  
             symbol={'naive_rolling': 'circle', 'louise_rolling': 'diamond'},  
             size=3.5,  
             colors={'naive_rolling': 'blue', 'louise_rolling': 'red'}, ),)

            louise_rolling  naive_rolling
Date                                     
1985-02-28             NaN            NaN
1985-03-31             NaN            NaN
1985-04-30             NaN            NaN
1985-05-31             NaN            NaN
1985-06-30             NaN            NaN
...                    ...            ...
2020-09-30       -0.176590       0.652793
2020-10-31       -0.414508       0.454900
2020-11-30       -0.035657       0.701709
2020-12-31       -0.142407       0.739108
2021-01-31       -0.154402       0.702721

[432 rows x 2 columns]


# Calculating Rolling Volatilities

In [11]:
df_n['naive_std'] = df_n['R-RF'].rolling(12).std() 
df_n['naive_std'].dropna()
volatile_plot = df_n[['Date','naive_std']]
volatile_plot.set_index('Date')

naive_std
Date                 
1985-02-28        NaN
1985-03-31        NaN
1985-04-30        NaN
1985-05-31        NaN
1985-06-30        NaN
...               ...
2020-09-30   7.076253
2020-10-31   7.139880
2020-11-30   7.683247
2020-12-31   7.699337
2021-01-31   7.718768

[432 rows x 1 columns]

Next, we analyse the rolling volatility which indicates that Louise's portfolio is less volatile than the naive strategy on average. This is owing to the fact that Louise tries to time the market and shifts her entire portfolio to risk-free debt at times removing any aspect of volatility. We notice this large difference in volatility in 2008 owing to the stock market crash. Even though Louise’s portfolio is less volatile, the naive strategy is better overall based on its Sharpe ratio, return rate and total investment growth.


In [12]:
louiseDf['louise_volatile'] = louiseDf['R-RF'].rolling(12).std() 
louiseDf['louise_volatile'].dropna()
plotvolatile = louiseDf[['Date','louise_volatile']]
plotdf.set_index("Date")
finalvolatile = pd.merge(plotvolatile,volatile_plot,on = 'Date',how = 'inner')
finalvolatile = finalvolatile.set_index("Date")
print(finalvolatile)
plyo.iplot(  
    finalvolatile[['naive_std','louise_volatile']].iplot(asFigure=True,
                                                 title='Rolling Volatility Comparison',  
             xTitle='Date',  
             yTitle='Volatility',
                                                 mode={'naive_std': 'lines+markers', 'louise_volatile': 'lines+markers'},  
             symbol={'naive_std': 'circle', 'louise_volatile': 'diamond'},  
             size=3.5,  
             colors={'naive_std': 'green', 'louise_volatile': 'orange'}, ),)

            louise_volatile  naive_std
Date                                  
1985-02-28              NaN        NaN
1985-03-31              NaN        NaN
1985-04-30              NaN        NaN
1985-05-31              NaN        NaN
1985-06-30              NaN        NaN
...                     ...        ...
2020-09-30         5.800951   7.076253
2020-10-31         5.758945   7.139880
2020-11-30         6.590549   7.683247
2020-12-31         6.544572   7.699337
2021-01-31         6.543006   7.718768

[432 rows x 2 columns]


# Investment Value Calculation

In [13]:
lump = 500
regular = 500

df_n['Value'] = lump * df_n['Return'].add(1).cumprod()

In [14]:
df_n['Naive Portfolio'] = np.nan

df_n['Investment'] = regular
iloc_Value2 = df_n.columns.get_loc('Naive Portfolio')
iloc_Invest = df_n.columns.get_loc('Investment')
iloc_Return = df_n.columns.get_loc('Return') 

df_n.iloc[0, iloc_Value2] = lump * (1 + df_n.iloc[0, iloc_Return]) + df_n.iloc[0, iloc_Invest]

for i in range(1, len(df_n)):
    df_n.iloc[i, iloc_Value2] = df_n.iloc[i-1, iloc_Value2] * (1 + df_n.iloc[i, iloc_Return]) + df_n.iloc[i, iloc_Invest]

In [15]:
df_n[['Value', 'Naive Portfolio', 'Investment', 'Return']]
naive_invest = df_n['Naive Portfolio']
naive_invest = pd.DataFrame(naive_invest)
naive_invest.index.name = 'Date'
print(naive_invest)

            Naive Portfolio
Date                       
1985-02-28     1.006905e+03
1985-03-31     1.498392e+03
1985-04-30     2.006952e+03
1985-05-31     2.628037e+03
1985-06-30     3.144520e+03
...                     ...
2020-09-30     1.681263e+06
2020-10-31     1.643481e+06
2020-11-30     1.823706e+06
2020-12-31     1.886961e+06
2021-01-31     1.875345e+06

[432 rows x 1 columns]


In [16]:
louise_invest = pd.DataFrame(currentBalLst,columns=['Louise Portfolio'])
louise_invest['Date'] = pd.date_range(start='19850228', periods=432, freq='M')
louise_invest = louise_invest.set_index('Date')
print(louise_invest)

            Louise Portfolio
Date                        
1985-02-28      5.069048e+02
1985-03-31      9.983916e+02
1985-04-30      1.506952e+03
1985-05-31      2.128037e+03
1985-06-30      2.645382e+03
...                      ...
2020-09-30      1.531557e+06
2020-10-31      1.497172e+06
2020-11-30      1.661452e+06
2020-12-31      1.662118e+06
2021-01-31      1.662784e+06

[432 rows x 1 columns]


Our naive strategy comprised Louise holding her portfolio in VFINX for the entire duration, she would have more money today than she has following her original strategy. If we analyse the chart below we see how the naive strategy gains significantly in the last year. 

Whenever we look at the value of an investment, we cannot ignore two events that are outliers and cannot be mapped by strategy, one would be the 2008 recession and the ongoing coronavirus pandemic. We see a sharp drop in the naive portfolio around 2008 as equities fell sharply and we see a sharp uprise in the naive portfolio in the last 6-8 months as the market has been really volatile and inflated. The market has been reaching all time highs and holding equities would prove very beneficial. Louise would have lost a lot of money when the pandemic started as the market crashed and following her strategy she would have moved her portfolio to the risk-free debt option and she would not have been able to enjoy the sharp recovery of the stock market.

In [17]:
finalinvest = pd.merge(louise_invest,naive_invest,on = 'Date',how = 'inner')
plyo.iplot(  
    finalinvest[['Louise Portfolio','Naive Portfolio']].iplot(asFigure=True,
                                                 title='Investment Value Comparison',  
             xTitle='Date',  
             yTitle='Investment Value',
                                                 mode={'Louise Portfolio': 'lines+markers', 'Naive Portfolio': 'lines+markers'},  
             symbol={'Louise Portfolio': 'circle', 'Naive Portfolio': 'diamond'},  
             size=2.75,  
             colors={'Louise Portfolio': 'red', 'Naive Portfolio': 'green'}, ),)


The figure below shows which strategy Louise was following in a given point of time. She clearly changed her strategy to risk-free debt in 2008 and a few times since the pandemic started.

In [18]:
portfolio_direction = pd.DataFrame(portfolioStocks,columns=['Indicator'])
portfolio_direction['Date'] = pd.date_range(start='19850228', periods=433, freq='M')
portfolio_direction = portfolio_direction.set_index('Date')
plyo.iplot(  
    portfolio_direction[['Indicator']].iplot(asFigure=True,
                                                 title='Strategy Change Points',  
             xTitle='Date',  
             yTitle='Strategy Point',
                                                 mode={'Indicator': 'markers'},  
             symbol={'Indicator': 'diamond'},  
             size=4.5,  
             colors={'Indicator': 'orange'}, ),)

# Returns Comparison

It is equally important to compare the return rate in both portfolios as well. If we analyse the chart below, we notice that there are a lot of high and low isolated blue points which indicate points where Louise switched to the risk-free strategy, we notice a lot of high points in the chart where Louise did not hold her portfolio in VFINX causing large losses. So overall, we notice that the overall return over time for the naive strategy is better owing to these high points (blue) in the chart.

In [19]:
vfinx_return = df_n.set_index('Date')
portfolio_return = pd.DataFrame(returnsLst,columns=['Louise Returns'])
portfolio_return['Date'] = pd.date_range(start='19850228', periods=432, freq='M')
portfolio_return = portfolio_return.set_index('Date')
portfolio_return = pd.merge(portfolio_return,vfinx_return, on = 'Date')
portfolio_return = pd.merge(portfolio_return,louise_invest, on = 'Date')
portfolio_return = portfolio_return[['Louise Returns','R','Naive Portfolio','Louise Portfolio']]
portfolio_return.columns = ['Louise Returns', 'Naive Returns','Naive Portfolio','Louise Portfolio']


In [20]:
plyo.iplot(  
    portfolio_return[['Louise Returns','Naive Returns']].iplot(asFigure=True,
                    title='Return Comparison',  
                     xTitle='Date',  
                     yTitle='Returns',
                    mode={'Louise Returns': 'lines+markers','Naive Returns' : 'markers'},  
             symbol={'Louise Returns': 'circle', 'Naive Returns' : 'diamond'},  
             size=5.5,  
             colors={'Louise Returns': 'orange', 'Naive Returns' : 'blue'}, ),)

In [21]:
vfinx_return = df_n.set_index('Date')
return_df = pd.merge(vfinx_return,gspc,on = 'Date')
return_df = return_df[['R','RF','Percent Return']]
return_df.columns = ['Vfinx Return','RF Return','S&P 500 Returns']
plyo.iplot(  
    return_df[['Vfinx Return','RF Return']].iplot(asFigure=True,
                    title='Return Comparison',  
                     xTitle='Date',  
                     yTitle='Returns',
                    mode={'Vfinx Return': 'lines+markers','RF Return' : 'lines+markers'},  
             symbol={'Vfinx Return': 'circle', 'RF Return' : 'diamond'},  
             size=2.5,  
             colors={'Vfinx Return': 'red', 'RF Return' : 'blue'}, ),)
plyo.iplot(  
    return_df[['S&P 500 Returns','RF Return']].iplot(asFigure=True,
                    title='Return Comparison',  
                     xTitle='Date',  
                     yTitle='Returns',
                    mode={'S&P 500 Returns': 'lines+markers','RF Return' : 'lines+markers'},  
             symbol={'S&P 500 Returns': 'circle', 'RF Return' : 'diamond'},  
             size=2.5,  
             colors={'S&P 500 Returns': 'orange', 'RF Return' : 'green'}, ),)
plyo.iplot(  
    return_df[['S&P 500 Returns','Vfinx Return']].iplot(asFigure=True,
                    title='Return Comparison',  
                     xTitle='Date',  
                     yTitle='Returns',
                    mode={'S&P 500 Returns': 'lines+markers','Vfinx Return' : 'lines+markers'},  
             symbol={'S&P 500 Returns': 'circle', 'Vfinx Return' : 'diamond'},  
             size=2.5,  
             colors={'S&P 500 Returns': 'blue', 'Vfinx Return' : 'magenta'}, ),)


In [22]:
vfinx_rf_corr = return_df['Vfinx Return'].corr(return_df['RF Return'])
sp500_rf_corr = return_df['S&P 500 Returns'].corr(return_df['RF Return'])
sp500_vfinx_corr = return_df['Vfinx Return'].corr(return_df['S&P 500 Returns'])
print('Vfinx vs RF = ',vfinx_rf_corr, 'S&P 500 vs RF = ',sp500_rf_corr,'S&P 500 vs VFINX = ',sp500_vfinx_corr)

Vfinx vs RF =  0.009056596806044984 S&P 500 vs RF =  0.02266933800486516 S&P 500 vs VFINX =  0.9826043213828569


# Table Generation

We have also printed tables that would further convince Louise that the naive strategy would have been more lucrative than her strategy. We start by showing her how much money she would have today compared to what she has today.


If she followed the naive strategy blindly, she would have approximately 1,875,345 assuming that she invests $ 500 in the beginning of every month. Based on her strategy she has approximately 1,662,784 today. So, following the naive strategy she would have approximately 212,561 more which is almost 13 % more than her current portfolio.

In [23]:
finalinvest.tail().round()

Louise Portfolio  Naive Portfolio
Date                                         
2020-09-30         1531557.0        1681263.0
2020-10-31         1497172.0        1643481.0
2020-11-30         1661452.0        1823706.0
2020-12-31         1662118.0        1886961.0
2021-01-31         1662784.0        1875345.0

To analyse how well Louise timed the market, we look at 5 best and worst months of Louise’s strategy and the naive strategy. 

In [24]:
best_5_louise = portfolio_return.sort_values(by=['Louise Returns'], ascending=False)
best_5_louise = pd.DataFrame(best_5_louise)
best_5_naive = portfolio_return.sort_values(by=['Naive Returns'], ascending=False)
best_5_naive = pd.DataFrame(best_5_naive)
worst_5_louise = portfolio_return.sort_values(by=['Louise Returns'], ascending=True)
worst_5_louise = pd.DataFrame(worst_5_louise)
worst_5_naive = portfolio_return.sort_values(by=['Naive Returns'], ascending=True)
worst_5_naive = pd.DataFrame(worst_5_naive)


First, we look at the best 5 months of Louise’s portfolio and we notice that Louise's strategy matches the returns of the naive strategy 

In [25]:
best_5_louise.head()

Louise Returns  Naive Returns  Naive Portfolio  Louise Portfolio
Date                                                                        
1987-01-31       14.864035      14.864035     1.502243e+04      1.416902e+04
2020-11-30       10.935640      10.935640     1.823706e+06      1.661452e+06
1990-05-31        9.692764       9.692764     4.622456e+04      4.445086e+04
1989-07-31        9.617159       9.617159     3.885174e+04      3.421009e+04
2000-03-31        9.526469       9.526469     4.038889e+05      3.048890e+05

Next, we look at the 5 best months of the naive strategy and we notice that in 3 out of the best 5 months of the Vanguard Fund, Louise had her investments at a risk-free rate based on her strategy which is a huge loss for her.


In [26]:
best_5_naive.head()

Louise Returns  Naive Returns  Naive Portfolio  Louise Portfolio
Date                                                                        
1987-01-31       14.864035      14.864035     1.502243e+04      1.416902e+04
2020-04-30        0.120000      13.271939     1.448965e+06      1.402832e+06
2011-10-31        0.000000      11.485867     4.994631e+05      6.970937e+05
2020-11-30       10.935640      10.935640     1.823706e+06      1.661452e+06
2009-04-30        0.020000      10.331337     3.199547e+05      5.009615e+05

Again, when we look at the worst 5 months of Louise’s strategy which notice that it is the same as the naive strategy which is actually a positive sign for Louise.


In [27]:
worst_5_louise.head()

Louise Returns  Naive Returns  Naive Portfolio  Louise Portfolio
Date                                                                        
1987-10-31      -21.288631     -21.288631     1.792459e+04      1.709377e+04
1998-08-31      -14.473561     -14.473561     2.487749e+05      2.412604e+05
2020-03-31      -12.755538     -12.755538     1.278750e+06      1.400651e+06
1986-12-31      -11.131487     -11.131487     1.264315e+04      1.183547e+04
2018-12-31       -9.501847      -9.501847     1.204502e+06      1.319902e+06

Lastly, when we look at the worst 5 months of Louise’s strategy we notice that Louise does well as she avoids 2 of the worst 5 months of the naive strategy.


In [28]:
worst_5_naive.head()

Louise Returns  Naive Returns  Naive Portfolio  Louise Portfolio
Date                                                                        
1987-10-31      -21.288631     -21.288631     1.792459e+04      1.709377e+04
2008-10-31        0.150000     -16.308157     3.462528e+05      4.972635e+05
1998-08-31      -14.473561     -14.473561     2.487749e+05      2.412604e+05
2020-03-31      -12.755538     -12.755538     1.278750e+06      1.400651e+06
2002-09-30        0.140000     -11.281377     2.369362e+05      3.258269e+05

Overall, we realise that due to her strategy Louise misses out on bigger gains than she avoids losses, so overall we come to the conclusion that Louise’s strategy is actually worse off as compared to naive strategy.

# Trade Offs

The major trade-off that has been discussed is the loss of money of approximately $210,000. Other than that trade-off, there are a few other trade-offs that Louise must consider when choosing between her strategy and the naive strategy. A few conceptual trade-offs with Louise's strategy is that she has to pay the fees to trade such big amounts every time. Furthermore, she would have to keep tracking the portfolio to stick to her strategy and time the market.

In [29]:
naive_vola = finalvolatile['naive_std'].mean()
louise_vola = finalvolatile['louise_volatile'].mean()
print('Average Rolling Naive Portfolio Volatility = ',naive_vola)
print('Average Rolling Louise Portfolio Volatility = ',louise_vola)

Average Rolling Naive Portfolio Volatility =  4.115847381792923
Average Rolling Louise Portfolio Volatility =  3.230471398884641


In [30]:
naive_return = portfolio_return['Naive Returns'].mean()*12
louise_return = portfolio_return['Louise Returns'].mean()*12
print('Annual Average Naive Portfolio Return',naive_return)
print('Annual Average Louise Portfolio Return',louise_return)

Annual Average Naive Portfolio Return 11.191332825020211
Annual Average Louise Portfolio Return 10.131528305133617


In switching to the naive strategy Louise will have to choose a strategy that is more risky as the volatility is higher (4.1% vs 3.2%). In choosing the Louise strategy the tradeoff is giving up a higher return percentage which is an opportunity to make higher returns (11.2% vs 10.1%)

In [31]:
new_df = finalinvest.loc['2010':'2019'].resample('Y').last().round(2)
new_df['Louise Returns'] = new_df['Louise Portfolio'].pct_change(1)
new_df['Naive Returns'] = new_df['Naive Portfolio'].pct_change(1)
new_df['Louise Returns Higher?'] = new_df['Louise Returns'] > new_df['Naive Returns']
new_df = pd.DataFrame(new_df)
new_df

Louise Portfolio  Naive Portfolio  Louise Returns  Naive Returns  \
Date                                                                           
2010-12-31         663989.07        486788.99             NaN            NaN   
2011-12-31         698093.73        501545.99        0.051363       0.030315   
2012-12-31         713673.17        586467.95        0.022317       0.169320   
2013-12-31         951732.25        783170.35        0.333569       0.335402   
2014-12-31        1086647.48        895268.37        0.141758       0.143134   
2015-12-31        1147146.23        912184.18        0.055675       0.018895   
2016-12-31        1179382.20       1026045.83        0.028101       0.124823   
2017-12-31        1409030.50       1256650.64        0.194719       0.224751   
2018-12-31        1319902.34       1204502.43       -0.063255      -0.041498   
2019-12-31        1740921.01       1589149.71        0.318977       0.319341   

            Louise Returns Higher?  
Date                                
2010-12-31                   False  
2011-12-31                    True  
2012-12-31                   False  
2013-12-31                   False  
2014-12-31                   False  
2015-12-31                    True  
2016-12-31                   False  
2017-12-31                   False  
2018-12-31                   False  
2019-12-31                   False

If we look at the yearly returns of the portfolio from 2010 to 2019, we notice that the returns are inconsistent. We notice that her strategy is better than the naive strategy in 2011 and 2015. If she completely moves to the passive strategy, she misses out on the opportunity to make higher returns in 2011 and 2015. 


In [32]:
naive_excess_returns = 12 * df_n['R-RF'].mean()
louise_excess_returns = 12 * (sum(rrfLst) / len(rrfLst))
print('Excess Returns Naive:',naive_excess_returns)
print('Excess Returns Louise:',louise_excess_returns)

Excess Returns Naive: 8.042999491686881
Excess Returns Louise: 6.965139416244739


When comparing Louise's strategy to the naive startegy, it is important to anaylze the excess returns of both strategies. If two investing strategies are compared, the one with higher excess returns is generally the one that performed better. It is clear that the naive startegy in this case had higher excess returns than Louise's. Lower excess returns are usually associated with less risk. This means that Louise is likely applying less risk with her investment startegy but this comes with the loss of profits. Also in this specific case, it is unlikely that a strategy as safe as investing in the S&P 500 carries more risk anyway regardless of the excess returns.

# Conclusion

Overall we concluded that the naive strategy is indeed better than Louise’s strategy based on the current data we have but the one component that is beyond the scope of this project is the analysis of the macro environment. Louise’s strategy was almost always ahead of the naive strategy until the coronavirus pandemic struck. Almost all trading strategies have failed in the midst of this pandemic and there has been not one effective strategy that could ideally time this extremely volatile market. As it stands, the naive strategy is better but we do believe that an analysis of the macro environment combined with these results might point in a different direction. 